### PR TITLE
comply with new osg 3.6.5 OQN API

### PR DESCRIPTION
**Pull request by mp3butcher**

### CODE REVIEW

**Comment by mp3butcher** [comment]
arf i missed initial bound...

**Comment by mp3butcher** [comment]
arf i can't tell if there's a difference in behavior with previous version...
Is it normal that sun doesn't flash when entering the frustum?

**Comment by AnyOldName3** [comment]
I see changes in the version less than path.

**Comment by mp3butcher** [comment]
> I see changes in the version less than path.

@AnyOldName3 i don't understand
this pr is simply the consequence of https://github.com/openscenegraph/OpenSceneGraph/pull/902 ....

**Comment by AnyOldName3** [comment]
A change has been made in OSG, so you've added a version check to do something different if a version with that change is present. However, you've changed the code that gets executed when the new change isn't present, too.

**Comment by mp3butcher** [comment]
>  However, you've changed the code that gets executed when the new change isn't present, too.
@AnyOldName3 
I only factorized qg->getQueryGeometry() in 3.4 path, that's not a big deal...No?

File: apps/openmw/mwrender/sky.cpp [hunk 1]
```diff
@@ -557,26 +558,33 @@ class Sun : public CelestialBody
...
         oqn->setQueriesEnabled(true);
-
+#if OSG_VERSION_LESS_THAN(3,6,5)
         // Make it fast! A DYNAMIC query geometry means we can't break frame until the flare is rendered (which is rendered after all the other geometry,
         // so that would be pretty bad). STATIC should be safe, since our node's local bounds are static, thus computeBounds() which modifies the queryGeometry
         // is only called once.
         // Note the debug geometry setDebugDisplay(true) is always DYNAMIC and that can't be changed, not a big deal.
-        oqn->getQueryGeometry()->setDataVariance(osg::Object::STATIC);
+        osg::Geometry* queryGeom = oqn->getQueryGeometry();
+        queryGeom->setDataVariance(osg::Object::STATIC);
```
Comment (AnyOldName3) [inline apps/openmw/mwrender/sky.cpp]: You're not setting the data variance to static in the 3.6.5 path.

**Pull request closed without merging**